src/Renderer/Color.cpp

```diff
@@ -48,6 +48,40 @@ const Color Color::NoAlpha(0, 0, 0, 0);
 Color::Color(uint8_t r, uint8_t g, uint8_t b, uint8_t a) : mR(r), mG(g), mB(b), mA(a)
 {}
 
+Color::Color(uint32_t value)
+{
+	setRgbaFromRawValue(value);
+}
+
+void Color::setRgbaFromRawValue(uint32_t value)
+{
+	mR = static_cast<uint8_t>((value & 0xFF000000u) >> 24);
+	mG = static_cast<uint8_t>((value & 0x00FF0000u) >> 16);
+	mB = static_cast<uint8_t>((value & 0x0000FF00u) >> 8);
+	mA = static_cast<uint8_t>((value & 0x000000FFu) >> 0);
```

Comment:
This actually seems like a reasonable assumption concerning endianess. It assumes red is in the higher order bits, and blue in the lower order bits. This is consistent with how hex color codes are typically written, such as in HTML. As they are translated, they are written to named fields. This seems like a reasonably safe thing to do.